src/irmin-pack/unix/pack_store.ml

```diff
@@ -420,22 +420,28 @@ struct
     Lwt.try_bind (fun () -> f (cast t)) on_success on_fail
 
   let unsafe_append ~ensure_unique ~overcommit t hash v =
+    let kind = Val.kind v in
+    let use_index =
```

Comment:
we also need to use the index when the strategy is always?

Reply:
yea this is completely buggy 😓  I've pushed a fix